### DISCUSSION TITLE
Adding support for uv!

### DISCUSSION THREAD

**Discussion by DavidVujic** [DavidVujic/python-polylith#255] [Announcements]
The Python tools for the Polylith Architecture now has full support for `uv`, a tool developed by the creators of `ruff`. 🎉 

Just like `Rye`, `uv` is using `Hatch` as the build backend. This means that the Polylith tool already had almost full support since before. With the Polylith CLI version **1.15.0**, reading the uv-specific lock file (the `uv.lock` file) has been added. That means you can use `uv` and expect the same kind of developer experience as `Rye`, `Hatch` and `PDM` when using the Polylith CLI.

### More information
The `uv` tool has its own set of commands, and you will find the docs about `uv` at the [Astral web site](https://docs.astral.sh/uv/).
There's a new example repo for **Polylith** and **uv**: https://github.com/DavidVujic/python-polylith-example-uv
The Polylith docs, updated with `uv`-specific info: https://davidvujic.github.io/python-polylith-docs/

**Comment by gwdekker** [comment]
what! amazing!

**Reply by gwdekker** [reply]
@DavidVujic I just want to let you know that your project of showing polylith to the python world has had great value to me and to the company I am with. I was looking for ways of doing monorepos in python and polylith is a great fit for us. Amazing your tooling and examples show all the main project tools (I would have dismissed polylith immediately if it would have forced me back to poetry). Now when the new large release of uv came out I was afraid I would have to choose between uv and polylith but that does not seem to be the case! Thanks a lot, your work is making a difference!
we are not switching from rye to uv just yet but will let you know how it went with polylith and uv when we do.

**Reply by DavidVujic** [reply]
Reading this makes me really happy, thank you for sharing! 😍 

Looking forward to hear about what you will do in the future.

After a while, maybe you also can consider sharing your setup to the Polylith docs by running `poly info` and screenshot or copy the output (`poly info  | pbcopy` if you are on mac)? I am looking for production systems using Polylith. If this is something you can and want to do, you could also add a short intro to the business in the same style as the ones already there: https://davidvujic.github.io/python-polylith-docs/examples/
